{
	"schema_version": "1.6.0",
	"id": "BIT-tensorflow-2022-35984",
	"modified": "2024-03-06T11:25:28.861Z",
	"published": "2024-03-06T11:13:25.198Z",
	"aliases": [
		"CVE-2022-35984",
		"GHSA-p2xf-8hgm-hpw5"
	],
	"details": "TensorFlow is an open source platform for machine learning. `ParameterizedTruncatedNormal` assumes `shape` is of type `int32`. A valid `shape` of type `int64` results in a mismatched type `CHECK` fail that can be used to trigger a denial of service attack. We have patched the issue in GitHub commit 72180be03447a10810edca700cbc9af690dfeb51. The fix will be included in TensorFlow 2.10.0. We will also cherrypick this commit on TensorFlow 2.9.1, TensorFlow 2.8.1, and TensorFlow 2.7.2, as these are also affected and still in supported range. There are no known workarounds for this issue.",
	"affected": [
		{
			"package": {
				"ecosystem": "Bitnami",
				"name": "tensorflow",
				"purl": "pkg:bitnami/tensorflow"
			},
			"severity": [
				{
					"type": "CVSS_V3",
					"score": "CVSS:3.1/AV:N/AC:L/PR:N/UI:N/S:U/C:N/I:N/A:H"
				}
			],
			"ranges": [
				{
					"type": "SEMVER",
					"events": [
						{
							"introduced": "0"
						},
						{
							"fixed": "2.7.2"
						},
						{
							"introduced": "2.8.0"
						},
						{
							"fixed": "2.8.1"
						},
						{
							"introduced": "2.9.0"
						},
						{
							"fixed": "2.9.1"
						}
					]
				}
			],
			"database_specific": {
				"source": "https://github.com/bitnami/vulndb/tree/main/data/tensorflow/BIT-tensorflow-2022-35984.json"
			}
		}
	],
	"references": [
		{
			"type": "WEB",
			"url": "https://github.com/tensorflow/tensorflow/commit/72180be03447a10810edca700cbc9af690dfeb51"
		},
		{
			"type": "WEB",
			"url": "https://github.com/tensorflow/tensorflow/security/advisories/GHSA-p2xf-8hgm-hpw5"
		}
	],
	"database_specific": {
		"cpes": [
			"cpe:2.3:a:google:tensorflow:*:*:*:*:*:*:*:*"
		],
		"severity": "High"
	}
}
